frame/staking/src/pallet/impls.rs

```diff
@@ -871,12 +879,15 @@ impl<T: Config> Pallet<T> {
 	/// to `Nominators` or `VoterList` outside of this function is almost certainly
 	/// wrong.
 	pub fn do_add_nominator(who: &T::AccountId, nominations: Nominations<T>) {
-		if !Nominators::<T>::contains_key(who) {
+		let prev_nominations = Self::nominations(who);
+		if !prev_nominations.is_none() {
 			// maybe update sorted list.
 			let _ = T::VoterList::on_insert(who.clone(), Self::weight_of(who))
 				.defensive_unwrap_or_default();
 		}
 		Nominators::<T>::insert(who, nominations);
+		let _ = T::EventListener::on_nominator_add(who, prev_nominations.unwrap_or_default())
+			.defensive();
```

Comment:
Why is this a defensive opration? It should be explained like `.defensive("EventListeners should never fail because of ...")`

Also, I am not sure if returning a result is actually reasonable here, if staking is not going to do anything about it. 

Looking at it from a blank slate, I think staking should not care about its event listeneres. It is set and forget.

Reply:
Yeah, I was thinking the same. It kinda made sense to me for a moment, but I guess if we really want things to fail hard - we could panic in the `EventListener`, but the initial idea was indeed to make sure that EventListeners don't mess with the Staking flow.
One argument for potentially handling results is the fact that Staking also relies on some of the listeners to work correctly, because they are maintaining sorted lists Staking is using, but since we can't know which one - let's get rid of result return types altogether and make this void.

Reply:
I have removed Result return types.